build/ci/official.yml

```diff
@@ -18,37 +21,61 @@ variables:
+# INSTALLATION
+###################################################################################################################################################################
+
 steps:
-- task: ms-vseng.MicroBuildTasks.30666190-6959-11e5-9f96-f56098202fef.MicroBuildSigningPlugin@1
+- task: MicroBuildSigningPlugin@4
   displayName: Install Signing Plugin
   inputs:
     signType: real
     esrpSigning: true
-    
-- task: ms-vseng.MicroBuildTasks.32f78468-e895-4f47-962c-58a699361df8.MicroBuildSwixPlugin@1
+
+# This is used as a PackageReference with ExcludeAssets="all", which means it is not executed.
+# RepoToolset has custom logic to run the contents of this plugin, so it is not executed normally.
+- task: MicroBuildSwixPlugin@1
   displayName: Install Swix Plugin
 
-- task: NuGetCommand@2
-  displayName: Restore 'Microsoft.DotNet.IBCMerge' package
+# This requires the OptimizationInputs via the 'Publish OptimizationInputs drop' step in the release pipeline.
+# To generate ProfilingInputs for the first time, set this input: ShouldSkipOptimize: true
+# This plugin will restore MicroBuild.Plugins.OptProf which is required to create ProfilingInputs data.
+# The projects include OptProf.targets which uses MicroBuild.Plugins.OptProf to run the target GenerateProfilingInputs on each project.
+- task: MicroBuildOptProfPlugin@6
+  displayName: Install OptProf Plugin
   inputs:
-    command: restore
-    feedsToUse: config
-    restoreSolution: 'build\proj\internal\RestoreIBCMerge.csproj'
-    nugetConfigPath: 'NuGet.config'
+    ProfilingInputsDropName: 'ProfilingInputs/$(System.TeamProject)/$(Build.Repository.Name)/$(Build.SourceBranchName)/$(Build.BuildId)'
+    OptimizationInputsLookupMethod: 'DropPrefix'
+    DropNamePrefix: 'OptimizationInputs/$(System.TeamProject)/$(Build.Repository.Name)'
+    AccessToken: '$(System.AccessToken)'
+    feedSource: 'https://devdiv.pkgs.visualstudio.com/DefaultCollection/_packaging/MicroBuildToolset/nuget/v3/index.json'
+    ShouldSkipOptimize: false
+    NumberCommitsToSearch: '100'
+
+###################################################################################################################################################################
+# BUILD SOLUTION
+###################################################################################################################################################################
 
-# TODO: Merged PIAs - reset to /ibc once private feeds are removed and Microsoft.DotNet.IBCMerge is reset
 - script: $(Build.SourcesDirectory)\build.cmd /build /test /ci /sign /diagnostic /no-deploy /no-integration /no-ibc /no-clearnugetcache /configuration $(BuildConfiguration)
   displayName: Build ProjectSystem.sln
 
-- task: Ref12.ref12-analyze-task.ref12-analyze-task.Ref12Analyze@0
-  displayName: Produce Codex index for http://index
+# Note: Does not verify that .vsix files are signed.
```

Comment:
Is that a limitation of `MicroBuildCodesignVerify` or a choice on our part?

Reply:
I don't actually know for sure, but I feel like it is a limitation. When you go to add this task, here are the options:

![image](https://user-images.githubusercontent.com/17788297/151459820-b30c133a-73ef-4eb2-b972-aa0b3b5af9a2.png)

If you click the `About this task` link, it only goes to our DevDiv wiki. The information for it is quite... malnourished: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/653/Codesign-Verification

The only setting I could look into is which folders we target. Maybe if I target a different folder, I can have it verify the `.vsix` file. But there don't seem to be settings on which file extensions the tool will verify. 🤔

Reply:
I added the folder in which we output the .vsix and now it is checking those files! It even checks that the files in the .vsix are signed properly. Run log here: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5686095&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=0881e0b9-84b3-5ca0-e860-5a43d1ab541c